### ISSUE TITLE
PR #208 causes FTBFS...

### ISSUE THREAD

**Issue by sunweaver** [ArcticaProject/nx-libs#219]
@uli42: I merged #208 too quickly and did not wait for my test build to finish. Is it possible that you forgot to add xkbfile.h in one of the #208 commits?

Could you please take another look at that? Or does it needed #214 to get merged, too.

Mike

```
rm -f log.o
gcc -c -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fPIE -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -Wall -Wpedantic -Wpointer-arith -Wundef    -I. -I../include -I../../../exports/include/nx-X11 -I../../../include/extensions         -I../../../programs/Xserver/Xext -I../../../programs/Xserver/render         -I../../../lib/Xau -I../../../../nxcomp         `pkg-config --cflags-only-I pixman-1`  -I../../.. -I../../../exports/include   -Dlinux -D__amd64__ -D_POSIX_C_SOURCE=199309L              -D_POSIX_SOURCE -D_XOPEN_SOURCE                 -D_DEFAULT_SOURCE                                 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64                 -D_GNU_SOURCE                  -DSHAPE     -DXINPUT    -DXKB   -DXCSECURITY    -DXF86BIGFONT   -DDPMSExtension         -DRENDER    -DRANDR     -DPANORAMIX     -DXFIXES    -DDAMAGE    -DCOMPOSITE      -DGCCUSESGAS -DAVOID_GLYPHBLT -DPIXPRIV -DSINGLEDEPTH              -DXvExtension                -DXFree86Server                -DXvMCExtension                 -DSMART_SCHEDULE                -DBUILDDEBUG -DXResExtension                -DX_BYTE_ORDER=X_LITTLE_ENDIAN              -DXORG_VERSION_CURRENT="(((6) * 10000000) + ((9) * 100000) + ((0) * 1000) + 0)" -D_XSERVER64 -DNDEBUG   -DFUNCPROTO=15 -DNARROWPROTO   -DUNIXCONN -DTCPCONN -DHAS_STICKY_DIR_BIT -DHAS_FCHOWN -DIPv6          -DHASXDMAUTH -DSECURE_RPC  -DDDXOSINIT -DSERVER_LOCK -DDDXOSFATALERROR                -DDDXOSVERRORF -DDDXTIME          -DUSE_RGB_TXT            -DBUILDERADDR='"xorg@lists.freedesktop.org"' -DFAIL_HARD -DNX_TRANS_SOCKET                     -DNX_TRANS_AUTH                       -DNX_TRANS_FOPEN                      -DNX_TRANS_SLEEP                      -DNX_TRANS_EXIT                       -DNX_TRANS_WAKEUP=1000   log.c
In file included from utils.c:147:0:
../include/xkbsrv.h:1100:21: fatal error: xkbfile.h: No such file or directory
 #include "xkbfile.h"
                     ^
compilation terminated.
Makefile:688: recipe for target 'utils.o' failed
make[9]: *** [utils.o] Error 1
make[9]: *** Waiting for unfinished jobs....
make[9]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11/programs/Xserver/os'
Makefile:748: recipe for target 'os' failed
make[8]: *** [os] Error 2
make[8]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11/programs/Xserver'
Makefile:507: recipe for target 'all' failed
make[7]: *** [all] Error 2
make[7]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11/programs'
xmakefile:509: recipe for target 'all' failed
make[6]: *** [all] Error 2
make[6]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11'
xmakefile:595: recipe for target 'World' failed
make[5]: *** [World] Error 2
make[5]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11'
Makefile:37: recipe for target 'World' failed
make[4]: *** [World] Error 2
make[4]: Leaving directory '/<<PKGBUILDDIR>>/nx-X11'
Makefile:54: recipe for target 'build-full' failed
make[3]: *** [build-full] Error 2
make[3]: Leaving directory '/<<PKGBUILDDIR>>'
Makefile:80: recipe for target 'build' failed
make[2]: *** [build] Error 2
make[2]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: make -j2 CDEBUGFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fPIE -fstack-protector-strong -Wformat -Werror=format-security LOCAL_LDFLAGS=-fPIE -pie -Wl,-z,relro -Wl,-z,now SHLIBGLOBALSFLAGS=-fPIE -Wl,-z,relro -Wl,-z,now SHLIBDIR=/usr/lib/x86_64-linux-gnu returned exit code 2
debian/rules:55: recipe for target 'override_dh_auto_build' failed
make[1]: *** [override_dh_auto_build] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
debian/rules:13: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

**Comment by uli42** [comment]
Yes, I seem to have forgotten xkbfile.h. Sigh. This is even worse, since the commit where it should have been included (2c31ce2f2a97a8dd358d95d3918c43d9474f3d16) has also a wrong commit message

So what's the best way to cope with this? Revert and re-commit?

Update: revert causes further grief, so I simply added a fix commit. Please retry!

**Comment by sunweaver** [comment]
Yes, revert that special commit and re-apply the correct version.

Mike

On Sat Oct 8 00:33:59 2016 GMT+0200, Ulrich Sibiller wrote:

> Yes, I seem to have forgotten xkbfile.h. Sigh. This is even worse, since the commit where it should have been included (2c31ce2) has also a wrong commit message 
> So what's the best way to cope with this? Revert and re-commit? 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.￼

## 

Sent from my Fairphone (running Sailfish OS)...

**Comment by uli42** [comment]
Revert was too intrusive, have done a fix commit.

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "Mike Gabriel" notifications@github.com
Gesendet: ‎08.‎10.‎2016 09:00
An: "ArcticaProject/nx-libs" nx-libs@noreply.github.com
Cc: "Ulrich Sibiller" uli42@gmx.de; "Assign" assign@noreply.github.com
Betreff: Re: [ArcticaProject/nx-libs] PR #208 causes FTBFS... (#219)

Yes, revert that special commit and re-apply the correct version. 

Mike 

On Sat Oct 8 00:33:59 2016 GMT+0200, Ulrich Sibiller wrote: 

> Yes, I seem to have forgotten xkbfile.h. Sigh. This is even worse, since the commit where it should have been included (2c31ce2) has also a wrong commit message 
> So what's the best way to cope with this? Revert and re-commit? 
> — 
> You are receiving this because you authored the thread. 
> Reply to this email directly, view it on GitHub, or mute the thread.￼ 

## 

Sent from my Fairphone (running Sailfish OS)...
